core/src/main/java/org/apache/iceberg/actions/BaseRewriteDataFilesAction.java

```diff
@@ -226,11 +226,16 @@ public RewriteDataFilesActionResult execute() {
           return TableScanUtil.planTasks(splitTasks, targetSizeInBytes, splitLookback, splitOpenFileCost);
         })
         .flatMap(Streams::stream)
+        .filter(task -> task.files().size() > 1)
         .collect(Collectors.toList());
 
+    if (combinedScanTasks.isEmpty()) {
+      return RewriteDataFilesActionResult.empty();
+    }
+
     List<DataFile> addedDataFiles = rewriteDataForTasks(combinedScanTasks);
-    List<DataFile> currentDataFiles = filteredGroupedTasks.values().stream()
-        .flatMap(tasks -> tasks.stream().map(FileScanTask::file))
+    List<DataFile> currentDataFiles = combinedScanTasks.stream()
```

Comment:
Is this change correct  ?   If we have a single file1 with 200MB, and the target file size is 128MB, then it will be planned into two combined tasks:  128MB, 72MB. Finally the `file1` will occur twice in the currentDataFiles  ?

Reply:
if there is a datafile which size > targetSizeInBytes ,I think it should not be rewrited , I open a new PR to deal this . 
https://github.com/apache/iceberg/pull/1762

Reply:
It's right that we may don't have to rewrite those files whose size greater than target file size.  But in this patch, I don't understand why we need to change it , could you revert this pls ?

Reply:
`filteredGroupedTasks` contains all the datafiles that meet the conditions of the query. `combinedScanTasks` removes some datafiles that are smaller than targetSizeInBytes but do not need to be compressed. If when constructing `currentDataFiles`,we use filteredGroupedTasks  instead of combinedScanTasks, those large files that do not need to be compressed will be deleted, resulting in results Incorrect.

When I doing this PR, I forgot to consider the datefile  whose size greater than the target file size. I think it is another issue, so I opened a new PR #1762 to fix it when  I found this issue. I think we should modify #1762 first, then modify this PR, or merge #1762 into  this PR ? what do you think?

Reply:
I think we should not rely on specific file-selection strategies here, because different selection strategies  may result in duplicated files for `currentDataFiles`.  The correct way is:  remove the duplicated files so that the `currentDataFiles` has correct files.

I saw the [here](https://github.com/apache/iceberg/pull/1704/files#diff-a330da6f6014d3154bb0979281e53f2b69c9e32117d0223bdf94d514a05cf559R264) has used the `set` to remove duplicated files.  So it's OK now.